FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -2480,4 +2482,123 @@ \subsubsection{Building a Random Forest}
     \item Combine the result of each tree to obtain the final decision by taking the average for regression or majority vote for classification
 \end{enumerate}
 
+\section{Debugging}
+
+\textbf{Motivation}
+\begin{enumerate}
+    \item You carefully split your data into training, validation and test set
+    \item You put aside your test set
+    \item You train your favorite learning algorithm on the training set
+    \item You get 20\% error on the validation set which is unacceptably high
+\end{enumerate}
+
+\subsection{Bias and Variance}
+Different sources of error can prevent supervised learning algorithms from generalizing beyond their training set. The \textbf{bias} results from false assumptions in the learning algorithm. High bias can cause an algorithm to miss the relevant structure between features and target. While the \textbf{variance} results from sensitivity to small fluctuations in the training set. High variance can cause an algorithm to model the random noise in the training data rather than the intended outputs.
+
+\begin{itemize}
+    \item A model with high bias and low variance \textbf{underfits} the data
+    \item A model with low bias and high variance \textbf{overfits} the data
+\end{itemize}
+
+Also refer to the figure \ref{fig:underoverfittingbiasvariance} in section \ref{sec:Regularisation}.
+
+\subsubsection{Under- and Overfitting in Classification}
+\noindent
+\textbf{Underfitting}
+\begin{itemize}
+    \item A straight line is just not complex enough to separate the classes
+    \item This model performs badly on both training and test data
+\end{itemize}
+
+\noindent
+\textbf{Overfitting}
+\begin{itemize}
+    \item This model optimizes to much with respect to the training data
+    \item It shows high performance on training but low performance on test data
+\end{itemize}
+
+\subsubsection{Under- and Overfitting in Regression}
+
+\begin{figure}[tbh!]
+    \centering
+    \includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/underfitting_overfitting_regression}
+    \caption{Underfitting and Overfitting in Regression}
+    \label{fig:underfittingoverfittingregression}
+\end{figure}
+
+\vspace{1em}
+
+\noindent
+\textbf{Underfitting}
+\begin{itemize}
+    \item A straight line is just not complex enough to approximate the data
+    \item This model performs badly on both training and test data
+\end{itemize}
+
+\noindent
+\textbf{Overfitting}
+\begin{itemize}
+    \item This model optimizes to much with respect to the training data
+    \item It shows high performance on training but low performance on test data
+\end{itemize}
+
```

Comment:
Do over- and underfitting in regression and classification really need to be separate if they're basically the same anyway?

Reply:
But it's not the same thing =/
One you are looking for a line to separate classes, another time to fit it to your data points